target/linux/mediatek/image/mt7622.mk

```diff
@@ -239,3 +239,19 @@ define Device/ubnt_unifi-6-lr-ubootmod
   ARTIFACT/bl31-uboot.fip := bl31-uboot ubnt_unifi-6-lr
 endef
 TARGET_DEVICES += ubnt_unifi-6-lr-ubootmod
+
+define Device/xiaomi_redmi-router-ax6s
+	DEVICE_VENDOR := Xiaomi
+	DEVICE_MODEL := Redmi Router AX6S
+	DEVICE_ALT0_VENDOR := Xiaomi
+  DEVICE_ALT0_MODEL := Router AX3200
+	DEVICE_DTS := mt7622-xiaomi-redmi-router-ax6s
+	DEVICE_DTS_DIR := ../dts
+	DEVICE_PACKAGES := kmod-mt7915e
+	KERNEL := kernel-bin | lzma
+	KERNEL_INITRAMFS_SUFFIX := -recovery.itb
+	KERNEL_INITRAMFS := kernel-bin | lzma | fit lzma $$(KDIR)/image-$$(firstword $$(DEVICE_DTS)).dtb with-initrd | pad-to 128k
+	IMAGES := sysupgrade.itb
+	IMAGE/sysupgrade.itb := append-kernel | fit lzma $$(KDIR)/image-$$(firstword $$(DEVICE_DTS)).dtb external-static-with-rootfs | pad-rootfs | append-metadata
```

Comment:
Are you sure you want to have the rootfs included in the uImage.FIT for the bootloader to verify (that's what `external-static-with-rootfs` does)? Wouldn't it be better to have at least rootfs and rootfs_data in UBI?

Reply:
Would it be better to have rootfs in the fit since firmware and firmware1 are 30M each, and then the entire ubi as overlay? I think the start of those two partitions would just end up being "hot" regardless with this layout.

Either way I'm not entirely sure of the image generation commands and their ramifications. Were you thinking something along the lines of append-ubi | append-rootfs and then ubiformat it onto the device?

I've removed that in the meantime.

Reply:
I was massively overthinking the image building variables. Much cleaner to just use the inherited values.

One odd thing with sysupgrade though, is nand_do_platform_check a bit wonky? It tries to use board name as path to extract the CONTROL file, but the sysupgrade tar has the device name instead? The check fails and is visible in CLI, but still allows flashing. In luci it's not even visible at all?